### PR TITLE
Programmable SIP - Initial draft

### PR DESCRIPTION
**Pull request by tbedford**
## Description

Initial draft based on information from Deniz Gokcen.

**NOTE:** Please review but don't merge until we get the go ahead.

## Review

Deniz Gokcen is happy with this.

### CODE REVIEW

File: _documentation/en/voice/sip/concepts/programmable-sip.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+``` text
+yourcompany.sip-eu.nexmo.com
+alice@yourcompany.sip-us.nexmo.com
+12345@yourcompany.sip-ap.nexmo.com
+```
+
+## Authentication - Access Control Lists
+
+IP addresses of your devices and endpoints can be configured to reach your Nexmo application. Nexmo will only accept calls from the IP address list that is configured for your domain. This method will enable you to whitelist your endpoints’ and devices’ IP Addresses per your application.
```
Comment (mheap) [inline _documentation/en/voice/sip/concepts/programmable-sip.md]: How do we configure ACLs?

File: _documentation/en/voice/sip/concepts/programmable-sip.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+Nexmo’s Programmable SIP product enables you to integrate your existing SIP Infrastructure with Nexmo’s powerful conversational communications platform. This integration will enable you easily connect to mobile, landline, SIP and WebRTC endpoints including browsers and mobile applications. It will also bring Voice API functionality such as multichannel recording, IVR, Text to Speech, web socket connectivity for AI integrations and the power of contextual conversations to your platform.
+
+![SIP Connect Workflow](/assets/images/workflow_sip_connect.png)
+
+## Nexmo SIP Domains
+
+A _Nexmo SIP Domain_ contains configuration you need to connect to Nexmo SIP endpoints to link to your Nexmo application.
+
+To route a SIP call to your Nexmo application, you need to create a unique Nexmo domain, such as `yourcompany`. The domain name will form the SIP URI, for example `sip:number@yourcompany.sip.nexmo.com`, and any calls routed to that SIP URI will be routed to your application NCCO `answer_url`.
```
Comment (mheap) [inline _documentation/en/voice/sip/concepts/programmable-sip.md]: How do you create a domain?

File: _documentation/en/voice/sip/concepts/programmable-sip.md (hunk 1 above)
Comment (denizgkcn) [reply _documentation/en/voice/sip/concepts/programmable-sip.md]: It'll be done manually until we release the SIP API to configure domains & ACLs. After SIP API is released Customer Dashboard can also consume this information

File: _documentation/en/voice/sip/concepts/programmable-sip.md (hunk 2 above)
Comment (denizgkcn) [reply _documentation/en/voice/sip/concepts/programmable-sip.md]: It'll be done manually until we release the SIP API to configure domains & ACLs. After SIP API is released Customer Dashboard can also consume this information

File: _documentation/en/voice/sip/concepts/programmable-sip.md (hunk 1 above)
Comment (tbedford) [reply _documentation/en/voice/sip/concepts/programmable-sip.md]: Added contact info on getting set up.

File: _documentation/en/voice/sip/concepts/programmable-sip.md (hunk 2 above)
Comment (tbedford) [reply _documentation/en/voice/sip/concepts/programmable-sip.md]: Added contact info on getting set up.

**Comment by tbedford** [comment]
@mheap Have your issues been addressed satisfactorily?
@denizgkcn Are we good to go?

**Comment by tbedford** [comment]
@denizgkcn - we are good to go on this end. Please let us know when you want this to go live. Many thanks.

**Pull request merged**